src/smt/abduction_solver.cpp

```diff
@@ -73,19 +73,21 @@ bool AbductionSolver::getAbduct(const std::vector<Node>& axioms,
   d_subsolver->setLogic(l);
   // assert the abduction query
   d_subsolver->assertFormula(aconj);
-  return getAbductInternal(axioms, abd);
+  d_axioms = axioms;
+  return getAbductInternal(abd);
 }
 
-bool AbductionSolver::getAbduct(const std::vector<Node>& axioms,
-                                const Node& goal,
-                                Node& abd)
+bool AbductionSolver::getAbductNext(Node& abd)
 {
-  TypeNode grammarType;
-  return getAbduct(axioms, goal, grammarType, abd);
+  // Since we are using the subsolver's check-sat interface directly, we
+  // simply call getAbductInternal again here. We assert that the subsolver
+  // is already initialized, which must be the case or else we are not in the
+  // proper SMT mode to make this call.
+  Assert(d_subsolver != nullptr);
+  return getAbductInternal(abd);
```

Comment:
How come nothing is added to the call to `getAbductInternal` that would filter out the previous solution? Is it worth documenting this?

Reply:
Sure, I can add documentation.  The reason is this is the default behavior for sygus subsolvers with multiple check-sat.